src/System.Private.Uri/tests/FunctionalTests/UriIpHostTest.cs

```diff
@@ -234,6 +234,34 @@ private void ParseBadIPv4Address(string badIpv4String)
 
         #region IPv6
 
+        [Theory]
+        [InlineData("fe80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("Fe80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("fE80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "%eth10")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "%")]
+        [InlineData("fe80::e077:c9a3:eeba:b8e9", "%\u30AF\u20E7")]
```

Comment:
@rmkerr actually special characters works well...i forget `%` escape char in test, however this behaviour is supported by rfc ` SHOULD contain only ASCII` is not a `MUST`.

Reply:
That sounds fine to me, if we allowed Unicode characters before we should continue to allow them. Thanks for updating the tests.